lib/base/defer.hpp

```diff
@@ -30,13 +30,21 @@ class Defer
 	inline
 	~Defer()
 	{
-		try {
-			m_Func();
-		} catch (...) {
-			// https://stackoverflow.com/questions/130117/throwing-exceptions-out-of-a-destructor
+		if (m_Func != nullptr) {
```

Comment:
Btw. wondering whether ` != nullptr` is actually necessary.

Reply:
Doesn't really make a difference, both does pretty much the same:
* https://en.cppreference.com/w/cpp/utility/functional/function/operator_cmp
* https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool